### PR TITLE
Run CI only on changes affecting *.rs files

### LINKED ISSUE #380: add path filter to CI workflow to only trigger on *.rs files
https://help.github.com/en/articles/workflow-syntax-for-github-actions#onpushpull_requestpaths

### PR DESCRIPTION
**Pull request by mh84**
Closes #380.

### CODE REVIEW

**Comment by SirWindfield** [comment]
Thanks for your contribution! would you mind adding the same to the Github Actions workflow files? The relevant documentation is linked here: #380 :)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 name: Continuous Integration
 
 on: [pull_request]
-
+  paths:
+    - '**.rs'
+
```
Comment (SirWindfield) [inline .github/workflows/ci.yml]:
```yaml
on:
  pull_request:
    paths:
    - '**.rs'
```

It probably has to be written this way, as `on: [pull_request]` directly assigns to `on` without allowing more children.

Comment (mh84) [reply .github/workflows/ci.yml]: Alright, that didn't was obvious out of reading the docs.

**Comment by SirWindfield** [comment]
Ye, just noticed that this is actually not completed right as our dependency bot won't trigger new builds now because it only changes toml files. I need to add an exception for those too, ups :) 

Thanks for your help!

**Pull request merged**